### ISSUE TITLE
[runtime] ttnn.isfinite returns FLOAT32 for a bf16 input; compiler types it BFLOAT16 (debug runtime asserts, #7930 still open in practice)

### ISSUE THREAD

**Issue by agobeljicTT** [tenstorrent/tt-mlir#9344]
### Summary
With `TT_RUNTIME_DEBUG=ON` the TTNN runtime asserts after `ttnn.isfinite` on a bf16 tensor:

```
DataType mismatch, expected BFLOAT16, got FLOAT32
DEBUG_ASSERT @ runtime/lib/ttnn/debug/debug_apis.cpp:46: expectedDataType == actualDataType
 --- tt::runtime::ttnn::operations::eltwise::unary::run(EltwiseUnaryOp const*, ProgramContext&)
```

The compiler lowers `ttir.isfinite` (i1 result) to `ttnn.isfinite` typed as the input dtype, bf16; the `::ttnn::isfinite` kernel hands back f32. Release runtimes do not check and the bits get reinterpreted downstream; the debug runtime (CI) aborts.

Same symptom as #7930, which was closed by #7916 (builder support only), so the runtime mismatch was never fixed.

### Reproduce
CI: https://github.com/tenstorrent/tt-mlir/actions/runs/35332409205/job/105564963558 (n300-llmbox, tracy), `tests/python/test_sdpa.py::test_sdpa_backward_decomposes[*-eager]`, 7 cases.

Minimal, on any debug-runtime build:
```python
import torch, tt_crank.torch
x = torch.randn(32, 64, dtype=torch.bfloat16).to("tt")
torch.isneginf(x)   # tt-crank eager: build_isneginf = isfinite -> logical_not -> lt -> logical_and
```
`ttir.isfinite` alone should be enough: any bf16 input to `ttnn.isfinite`.

### Where it bites
torch's MATH decomposition of SDPA backward (`_safe_softmax`) calls `aten::isneginf`, so every eager training call that falls outside the ttml sdpa kernels (float mask, per-batch mask, S % 32 != 0, Sq != Sk, 3-D q/k/v) trips it. #9323 xfails those eager cases with `reason="#9344"` until this is fixed.

### Proposed fix
Either make `runEltwiseUnaryOp` for `IsFinite` (and the other predicate unaries: isinf/isnan/isposinf/isneginf if added) typecast the kernel output to the flatbuffer's expected dtype (`ttnn::typecast`, like the existing typecast handling in `unary.cpp`), or have TTIR->TTNN emit the typecast explicitly. Add a lit/golden test that runs under `TT_RUNTIME_DEBUG=ON`.

Refs: `runtime/lib/ttnn/operations/eltwise/unary/unary.cpp` (`case IsFinite`), `lib/Conversion/TTIRToTTNN/TTIRToTTNN.cpp` (`ElementwiseOpConversionPattern<ttir::IsFiniteOp, ttnn::IsFiniteOp>`), `tt-crank/src/engine/ttir_module_builder.cpp` (`build_isneginf`).
